### PR TITLE
Remove in-memory nav config cache

### PR DESCRIPTION
**Pull request by sobakasu**
workaround for simple navigation bug. rendering nav causes nav options to be deleted, including :if lambda procs. AIT uses nav :if lambda procs to hide nav items depending on whether a user is logged in. AIT renders two navigations (desktop and mobile). The second nav to be rendered (mobile) does not include the :if lambda procs. So the mobile nav shows items it shouldn't.

This change removes the in-memory cache of navigation configuration, so options are regenerated each time navigation is rendered.

### CODE REVIEW

File: app/helpers/koi/navigation_helper.rb [hunk 1]
```diff
@@ -27,16 +27,14 @@ def get_nav_items(key)
       Rails.cache.fetch(prefix_cache_key(key), expires_in: cache_expiry) do
         NavItem.navigation(key, binding())
       end
+    elsif Koi::Caching.nav_memory_caching
+      @get_nav_items ||= {}
+      @get_nav_items[key] ||= NavItem.navigation(key, binding())
```
Comment (sfnelson) [inline app/helpers/koi/navigation_helper.rb:32]: as discussed, this cache is not valuable because we can use Koi caching or Rails query caching, no need for the controller-scoped caching

File: app/models/nav_item.rb [hunk 2]
```diff
@@ -120,11 +120,8 @@ def draggable?
   end
 
   def navigation(get_binding=binding())
-    # FIXME: Caching procs and lambda causes "no marshal_dump is defined for class Proc"
-    # @nav_item ||= Rails.cache.fetch("nav_item/#{self.id}-#{self.updated_at}/navigation", expires_in: 7.days) do
-      @@binding = get_binding
-      children.collect { |c| c.to_hash unless c.is_hidden }.compact.flatten
-    # end
+    @@binding = get_binding
```
Comment (sfnelson) [inline app/models/nav_item.rb:123]: why did you remove the comment here?

Comment (sobakasu) [reply app/models/nav_item.rb]: removed this unrelated change

File: app/helpers/koi/navigation_helper.rb (hunk 1 above)
Comment (sobakasu) [reply app/helpers/koi/navigation_helper.rb]: removed this cache of navigation options
